### PR TITLE
Support windows development with docker

### PR DESCRIPTION
**Pull request by cailyncodes**
Fully support Docker development on Windows with only the addition of an environment variable.

(This isn't a perfect solution, of course, but it provides a consistent experience across all platforms. And is much better than previous solutions.)

### CODE REVIEW

**Comment by cmgiven** [comment]
I'm going to take myself off this and leave this to Julia if that's alright. I think her feedback will be far more useful than mine!

File: docker-compose-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - 5000:5000
+    entrypoint:
+      - sh
+      - /app/entrypoint.sh
+    depends_on:
+      - db
+  client:
+    build:
+      context: .
+      dockerfile: ./ClientApp/Dockerfile
```
Comment (jlhogan) [inline docker-compose-windows.yaml]: for consistency should this be named Dockerfile-dev as well ?

File: dc-dotnet [hunk 2]
```diff
@@ -1 +1,5 @@
-docker-compose exec backend dotnet $@
+if [ "$DOCKER_ON_WINDOWS" ]; then
```
Comment (jlhogan) [inline dc-dotnet]: also for consistency should this wrapper just be around `docker-compose exec backend` like the new `dc-client` is ? that seems like a more intuitive interaction. I guess it doesn't really matter much; i'm not clear on if OEC devs would ever use this docker-compose set up to develop?

File: ClientApp/entrypoint.sh [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
 nm=$(ls -A ./node_modules/*)
 if [ -z "$nm" ]; then
     echo "installing yarn dependencies..."
     yarn install
 fi
 
-echo "starting client..."
 if [ "$DOCKER_ON_WINDOWS" ]; then
-    CHOKIDAR_USEPOLLING=true CHOKIDAR_INTERVAL=2500 yarn start
+    rsync -aq --exclude node_modules /home/node/mapped-app/* /home/node/app > /dev/null 2>& 1
```
Comment (jlhogan) [inline ClientApp/entrypoint.sh]: can you capture the necessity of this in comment somewhere? To make it clear what the issue was, and how this solves it?

File: dc-dotnet (hunk 2 above)
Comment (cailyncodes) [reply dc-dotnet]: I'm good with either approach -- I was trying to minimize the amount that I changed the interface you had set up, but can go entirely consistent with one approach.

**Comment by cailyncodes** [comment]
Oops yeah totally! Once I got the npm modules thing working, I excitedly pushed, forgetting to add the comment in.

**Comment by cailyncodes** [comment]
Idea for `rsync` based solution inspired by https://github.com/docker/for-win/issues/969. Addition comments added to `entrypoint.sh` for client container.

**Pull request merged**